### PR TITLE
More bank 3 routines, plus teleport/spawn

### PR DESCRIPTION
**Pull request by Pokeglitch**

### CODE REVIEW

File: engine/events/overworld.asm [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	dw .FailTeleport
+	dw .CheckIfSpawnPoint
+
+.TryTeleport: ; 03:52fc
+	call GetMapEnvironment
+	cp TOWN
+	jr z, .success
+	cp ROUTE
+	jr z, .success
+	ld a, $02
```
Comment (mid-kid) [inline engine/events/overworld.asm]: Create constants for these, even if they're just MOVE_SCRIPT_02 or such. (Please check if pokecrystal has something similar)

File: engine/events/overworld.asm [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	ld [wFieldMoveScriptID], a
+	ret
+.success
+	ld a, $03
```
Comment (mid-kid) [inline engine/events/overworld.asm]: Ditto.

File: engine/events/overworld.asm [hunk 3]
```diff
@@ -0,0 +1,100 @@
+INCLUDE "constants.asm"
+
+SECTION "TeleportFunction", ROMX[$52db], BANK[$03]
+
+TeleportFunction: ; 03:52db
+	xor a
+	ld [wFieldMoveScriptID], a
+.loop
+	ld a, [wFieldMoveScriptID]
+	bit 7, a
```
Comment (mid-kid) [inline engine/events/overworld.asm]: This bit means something. What does it mean? Make a constant. Constantize all of its uses by or-ing it with the script ID value.

File: engine/overworld/spawn_points.asm [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	jr z, .fail
+	cp d
+	jr nz, .next
+	inc hl
+	ld a, [hld]
+	cp e
+	jr z, .succeed
+.next
+	push bc
+	ld bc, 4
```
Comment (mid-kid) [inline engine/overworld/spawn_points.asm]: You may ignore this to your own better judgement, but maybe it's a good idea to constantize this, and fix it in pokecrystal as well.

**Comment by Pokeglitch** [comment]
Added the constants as requested

**Pull request merged**